app/models/response_map.rb

```diff
@@ -47,8 +64,56 @@ def self.assessments_for(team)
     responses
   end
 
-  # Check to see if this response map is a survey. Default is false, and some subclasses will overwrite to true.
   def survey?
     false
   end
+
+  # Best-effort timestamp of when the reviewee (or their team) last touched the work.
+  def latest_submission_at_for_reviewee
+    return nil unless reviewee
+
+    candidates = []
+    candidates << reviewee.updated_at if reviewee.respond_to?(:updated_at) && reviewee.updated_at.present?
+
+    # Check team-related timestamps if the reviewee has a team
+    if reviewee.respond_to?(:team) && reviewee.team
+      team = reviewee.team
+      candidates << team.updated_at if team.respond_to?(:updated_at) && team.updated_at.present?
+
+    # Also check teams_participants or teams_users join records if they exist
+      if team.respond_to?(:teams_participants)
+        team.teams_participants.each do |tp|
+          candidates << tp.updated_at if tp.respond_to?(:updated_at) && tp.updated_at.present?
+        end
+      end
+    # Also check teams_users join records if they exist
+      if team.respond_to?(:teams_users)
+        team.teams_users.each do |tu|
```

Comment:
There is no teams_users in the new system!  It is teams_participants.  This may affect the logic.